Gems/Stars/Code/Source/StarsComponentController.cpp

```diff
@@ -0,0 +1,212 @@
+            }
+        }
+
+        OnConfigChanged();
+    }
+
+    void StarsComponentController::UnregisterFeatureProcessor()
+    {
+        if (m_scene && m_starsFeatureProcessor)
+        {
+            m_scene->DisableFeatureProcessor<StarsFeatureProcessor>();
+
+            m_starsFeatureProcessor = nullptr;
+
+            RPI::FeatureProcessorFactory::Get()->UnregisterFeatureProcessor<StarsFeatureProcessor>();
+        }
+    }
+
+    void StarsComponentController::OnStarsAssetChanged()
+    {
+        Data::AssetBus::MultiHandler::BusDisconnect();
+        if (m_configuration.m_starsAsset.GetId().IsValid())
+        {
+            Data::AssetBus::MultiHandler::BusConnect(m_configuration.m_starsAsset.GetId());
+            m_configuration.m_starsAsset.QueueLoad();
+        }
+    }
+
+    void StarsComponentController::OnAssetReady(Data::Asset<Data::AssetData> asset)
+    {
+        UpdateStarsFromAsset(asset);
+    }
+
+    void StarsComponentController::OnAssetReloaded(Data::Asset<Data::AssetData> asset)
+    {
+        UpdateStarsFromAsset(asset);
+    }
+
+    void StarsComponentController::UpdateStarsFromAsset(Data::Asset<Data::AssetData> asset)
+    {
+        if (asset.GetId() != m_configuration.m_starsAsset.GetId())
+        {
+            return;
+        }
+
+        m_configuration.m_starsAsset = asset;
+
+        StarsAsset *starsAsset = asset.GetAs<StarsAsset>();
+        if (starsAsset && starsAsset->m_data.size() > StarsAsset::HeaderSize && m_starsFeatureProcessor)
+        {
+            Star star{ 0 };
+            AZStd::array<float,3> position;
+            constexpr int verticesPerStar{ 6 };
+            const size_t starsAssetSize = starsAsset->m_data.size();
+            const size_t starSize = sizeof(star);
+            const size_t numStars = (starsAssetSize - StarsAsset::HeaderSize) / starSize;
+
+            AZStd::vector<StarsFeatureProcessorInterface::StarVertex> stars;
+            stars.resize(numStars * verticesPerStar);
+
+            IO::MemoryStream stream(starsAsset->m_data.data(), starsAssetSize);
+
+            // skip the header
+            stream.Seek(AZ::IO::OffsetType(StarsAsset::HeaderSize), IO::GenericStream::SeekMode::ST_SEEK_BEGIN);
+
+            for (uint32_t i = 0; i < numStars; ++i)
+            {
+                stream.Read(starSize, &star);
+                
+                position[0] = -cosf(AZ::DegToRad(star.declination)) * sinf(AZ::DegToRad(star.ascension * 15.0f));
+                position[1] = cosf(AZ::DegToRad(star.declination)) * cosf(AZ::DegToRad(star.ascension * 15.0f));
+                position[2] = sinf(AZ::DegToRad(star.declination));
+
+                for (int k = 0; k < verticesPerStar; k++)
+                {
+                    stars[6 * i + k].m_position = position;
+                    stars[6 * i + k].m_color = (star.magnitude << 24) + (star.blue << 16) + (star.green << 8) + star.red;
```

Comment:
Would be better to change the indexing to use `verticesPerStar` instead of hardcoding to 6. Currently, updating `verticesPerStar` on line 144 would break this.